src/dataLayer/databaseQueryManager.js

```diff
@@ -48,6 +50,12 @@ class DatabaseQueryManager {
          * @private
          */
         this._useAtlas = useAtlas;
+        /**
+         * @type {ResourceLocator}
+         */
+        this.resourceLocator = resourceLocatorFactory.createResourceLocator(this._resourceType,
+            this._base_version, this._useAtlas);
+        assert(this.resourceLocator instanceof ResourceLocator);
```

Comment:
Should this be in a unit test on the factory/container instead?  We should be able to trust what is being returned from the factory containers.

Reply:
It is easy to accidentally send the wrong object to the wrong parameter so having an extra check is helpful.